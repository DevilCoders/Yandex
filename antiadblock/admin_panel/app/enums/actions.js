export const ACTIONS = {
    // Временно скрыт мониторинг на сервисах, т.к. он пока не работает
    // GRAPHS: 'graphs',
    CONFIGS: 'configs',
    AUDIT: 'audit',
    INFO: 'info',
    UTILS: 'utils',
    // ANTIADB-2636
    // HEALTH: 'health',
    RULES: 'rules',
    SBS_RESULT_CHECKS: 'sbs-result-checks'
};

export const SECRET_ACTIONS = {
    SCREENSHOTS: 'screenshots-checks',
    PROFILES: 'sbs-profiles',
    LOGS: 'logs'
};
