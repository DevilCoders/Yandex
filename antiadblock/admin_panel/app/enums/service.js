export const STATUS = {
    ACTIVE: 'ok',
    INACTIVE: 'inactive'
};

export const STATUS_COLORS = {
    [STATUS.ACTIVE]: 'green',
    [STATUS.INACTIVE]: 'gray'
};

export const SUPPORTT_PRIORITY = {
    CRITICAL: 'critical',
    MAJOR: 'major',
    MINOR: 'minor',
    OTHER: 'other'
};
