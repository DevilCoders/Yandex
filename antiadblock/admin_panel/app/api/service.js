import {antiadbRequest} from 'app/lib/api';
import {METRICS} from 'app/enums/graphs';

const enumSbsRunCheck = {
    onlyMyRuns: 'only_my_runs',
    offset: 'offset',
    limit: 'limit',
    fromDate: 'from',
    toDate: 'to',
    sortedBy: 'sortedby'
};

export default {
    fetchService: function (serviceId) {
        return antiadbRequest({
            path: `/service/${serviceId}`,
            options: {
                method: 'GET'
            }
        });
    },

    fetchTrend: function (serviceId) {
        return antiadbRequest({
            path: `/service/${serviceId}/trend_ticket`,
            options: {
                method: 'GET'
            },
            queryParams: {
                type: 'negative_trend'
            }
        });
    },

    createTrend: function (serviceId, datetime) {
        return antiadbRequest({
            path: `/service/${serviceId}/create_ticket`,
            options: {
                method: 'POST',
                body: {
                    datetime: Math.floor((new Date(datetime)).getTime() / 1000),
                    type: 'negative_trend'
                }
            }
        });
    },

    fetchLabels: function (serviceId) {
        return antiadbRequest({
            path: '/labels',
            options: {
                method: 'GET'
            },
            queryParams: {
                service_id: serviceId
            }
        });
    },

    changeLabelParent: function (labelId, parentId) {
        return antiadbRequest({
            path: `/label/${labelId}/change_parent_label`,
            options: {
                method: 'POST',
                body: {
                    parent_label_id: parentId
                }
            }
        });
    },

    createLabel: function (labelId, parentId) {
        return antiadbRequest({
            path: '/label',
            options: {
                method: 'POST',
                body: {
                    label_id: labelId,
                    parent_label_id: parentId
                }
            }
        });
    },

    createService: function (serviceId, name, domain) {
        return antiadbRequest({
            path: '/service',
            options: {
                method: 'POST',
                body: {
                    service_id: serviceId,
                    name,
                    domain
                }
            }
        });
    },

    changePriority: function (serviceId, priority) {
        return antiadbRequest({
            path: `/service/${serviceId}/support_priority`,
            options: {
                method: 'PATCH',
                body: {
                    support_priority: priority
                }
            }
        });
    },

    fetchSchema: function (labelId) {
        return antiadbRequest({
            path: `/label/${labelId}/config/schema`,
            options: {
                method: 'GET'
            }
        });
    },

    fetchServiceConfigs: function (serviceId, offset, limit, filters) {
        return antiadbRequest({
            path: `/label/${serviceId}/configs`,
            options: {
                method: 'GET'
            },
            queryParams: {
                offset,
                limit,
                ...filters
            }
        });
    },

    fetchServiceMetric: function (serviceId, name, range) {
        const queryParams = {
            date_range: range
        };

        // Для метрики по таймингам дополнительный параметр - массив ожидаемых перцентилей
        if (name === METRICS.TIMINGS) {
            queryParams.percentile = [95, 98];
        }

        return antiadbRequest({
            path: `/metrics/${serviceId}/${name}`,
            options: {
                method: 'GET'
            },
            queryParams
        });
    },

    fetchServiceConfig: function (serviceId, configId) {
        return antiadbRequest({
            path: `/service/${serviceId}/config/${configId}`,
            options: {
                method: 'GET'
            }
        });
    },

    applyServiceConfig: function (serviceId, configId, options) {
        return antiadbRequest({
            path: `/service/${serviceId}/config/${configId}/${options.target}`,
            options: {
                method: 'PUT',
                body: {
                    old_id: options.oldConfigId ? options.oldConfigId : null
                }
            }
        });
    },

    applyExperiment: function (configId, options) {
        return antiadbRequest({
            path: `/config/${configId}/experiment`,
            options: {
                method: 'PATCH',
                body: {
                    exp_id: options.comment
                }
            }
        });
    },

    removeExperiment: function (configId) {
        return antiadbRequest({
            path: `/config/${configId}/experiment/remove`,
            options: {
                method: 'PATCH'
            }
        });
    },

    createServiceConfig: function (serviceId, parentId, comment, data) {
        return antiadbRequest({
            path: `/service/${serviceId}/config`,
            options: {
                method: 'POST',
                body: {
                    parent_id: parentId,
                    comment,
                    data
                }
            }
        });
    },

    fetchServiceSetup: function (serviceId) {
        return antiadbRequest({
            path: `/service/${serviceId}/setup`,
            options: {
                method: 'GET'
            }
        });
    },

    fetchServiceHealth(serviceId) {
        return antiadbRequest({
            path: `/service/${serviceId}/get_service_checks`,
            options: {
                method: 'GET'
            }
        });
    },

    fetchServiceAudit: function (serviceId, offset, limit, labelId) {
        return antiadbRequest({
            path: `/audit/service/${serviceId}`,
            options: {
                method: 'GET'
            },
            queryParams: {
                offset,
                limit,
                label_id: labelId
            }
        });
    },

    setServiceHealtInProgress(serviceId, checkId, hours) {
        return antiadbRequest({
            path: `/service/${serviceId}/check/${checkId}/in_progress`,
            options: {
                method: 'PATCH',
                body: {
                    hours
                }
            }
        });
    },

    decryptLink: function (serviceId, encrypted) {
        return antiadbRequest({
            path: `/service/${serviceId}/utils/decrypt_url`,
            options: {
                method: 'POST',
                body: {
                    url: encrypted
                }
            }
        });
    },

    disableService: function (serviceId) {
        return antiadbRequest({
            path: `/service/${serviceId}/disable`,
            options: {
                method: 'POST'
            }
        });
    },

    enableService: function (serviceId) {
        return antiadbRequest({
            path: `/service/${serviceId}/enable`,
            options: {
                method: 'POST'
            }
        });
    },

    generateToken: function (serviceId) {
        return antiadbRequest({
            path: `/service/${serviceId}/gen_token`,
            options: {
                method: 'GET'
            }
        });
    },

    addServiceComment: function (serviceId, comment) {
        return antiadbRequest({
            path: `/service/${serviceId}/comment`,
            options: {
                method: 'POST',
                body: {
                    comment
                }
            }
        });
    },

    getServiceComment: function (serviceId) {
        return antiadbRequest({
            path: `/service/${serviceId}/comment`,
            options: {
                method: 'GET'
            }
        });
    },

    fetchServiceRules: function (serviceId, tags = []) {
        return antiadbRequest({
            path: `/service/${serviceId}/sonar/rules`,
            options: {
                method: 'GET'
            },
            queryParams: {...(tags.length ? {tags: tags.join(',')} : {})}
        });
    },

    fetchServiceSbsProfile: function (serviceId, reqParams) {
        const params = reqParams || {};
        // тут profileId опциональный параметр, если он есть, то бек возвращает профиль с указанным id
        // если его нет, то возвращает текущий рабочий (последний)
        return antiadbRequest({
            path: `/service/${serviceId}/sbs_check/profile${params.profileId ? `/${params.profileId}` : ''}`,
            options: {
                method: 'GET'
            },
            queryParams: {
                tag: params.tag
            }
        });
    },

    deleteServiceSbsProfileByTag: function (serviceId, tag) {
        return antiadbRequest({
            path: `/service/${serviceId}/sbs_check/tag/${tag}`,
            options: {
                method: 'PATCH'
            }
        });
    },

    fetchServiceSbsProfileByTag: function (serviceId, tag) {
        return antiadbRequest({
            path: `/service/${serviceId}/sbs_check/tag/${tag}`,
            options: {
                method: 'GET'
            }
        });
    },

    fetchServiceSbsProfileTags: function (serviceId) {
        return antiadbRequest({
            path: `/service/${serviceId}/sbs_check/tags`,
            options: {
                method: 'GET'
            }
        });
    },

    fetchServiceSbsProfileTag: function (serviceId, tagName) {
        return antiadbRequest({
            path: `/service/${serviceId}/sbs_check/tag/${tagName}`,
            options: {
                method: 'GET'
            }
        });
    },

    changeServiceSbsProfileTag: function (serviceId, tagName, data) {
        return antiadbRequest({
            path: `/service/${serviceId}/sbs_check/tag/${tagName}`,
            options: {
                method: 'PATCH',
                body: {
                    data
                }
            }
        });
    },

    saveServiceSbsProfile: function (serviceId, data, tag) {
        return antiadbRequest({
            path: `/service/${serviceId}/sbs_check/profile`,
            options: {
                method: 'POST',
                body: {
                    data,
                    tag
                }
             }
        });
    },

    fetchServiceSbsCheckScreenshots: function (serviceId, resultId) {
        return antiadbRequest({
            path: `/service/${serviceId}/sbs_check/results/${resultId}`,
            options: {
                method: 'GET'
            }
        });
    },

    fetchSbsLogs: function (resId, reqId, type) {
        return antiadbRequest({
            path: `/sbs_check/results/${resId}/logs/${type}/id/${reqId}`,
            options: {
                method: 'GET'
            }
        });
    },

    fetchServiceSbsResultChecks: function (serviceId, options = {}) {
        const queryParams = Object.keys(enumSbsRunCheck).reduce((res, key) => {
            if (options[key] || Number.isFinite(options[key])) {
                res[enumSbsRunCheck[key]] = options[key];
            }

            return res;
        }, {});

        if (options.sortedBy) {
            queryParams.order = options.isReverseSorted ? 'desc' : 'acs';
        }

        return antiadbRequest({
            path: `/service/${serviceId}/sbs_check/results`,
            options: {
                method: 'GET'
            },
            queryParams
        }, 2);
    },

    runServiceSbsChecks: function (serviceId, testing, expId, tag) {
        const body = {};

        if (testing) {
            body.testing = testing;
        }

        if (expId) {
            body.id = expId;
        }

        if (tag) {
            body.tag = tag;
        }

        return antiadbRequest({
            path: `/service/${serviceId}/sbs_check/run`,
            options: {
                method: 'POST',
                body
            }
        });
    },

    setServiceEnableMonitoring: function (serviceId, monitoringName, monitoringIsEnable) {
        const key = `${monitoringName}_enabled`;

        return antiadbRequest({
            path: `/service/${serviceId}/set`,
            options: {
                method: 'POST',
                body: {
                    [key]: monitoringIsEnable
                }
            }
        });
    }
};
