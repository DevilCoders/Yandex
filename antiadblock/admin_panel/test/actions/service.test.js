import * as serviceActions from 'app/actions/service';
import * as serviceActionsFixtures from '../fixtures/actions/service';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import * as storeFixtures from '../fixtures/store';
import {SERVICE, SERVICE_CONFIG, SERVICE_CONFIGS, SETUP, SERVICE_AUDIT, SERVICE_CONFIG_ARCHIVED} from '../fixtures/backend/service';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('service actions', () => {
    it('should create START_SERVICE_LOADING', () => {
        expect(serviceActions.startServiceLoading(123)).toEqual(serviceActionsFixtures.START_SERVICE_LOADING);
    });

    it('should create END_SERVICE_LOADING', () => {
        expect(serviceActions.endServiceLoading(123, SERVICE)).toEqual(serviceActionsFixtures.END_SERVICE_LOADING);
    });

    it('should create START_SERVICE_LOADING and END_SERVICE_CONFIG_LOADING', () => {
        const store = mockStore(storeFixtures.SERVICE_EMPTY);

        fetch.mockResponseOnce(JSON.stringify(SERVICE));

        return store.dispatch(serviceActions.fetchService(123)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        });
    });

    it('should create START_SERVICE_LOADING and SET_ERRORS', () => {
        const store = mockStore(storeFixtures.SERVICE_EMPTY);

        fetch.mockRejectOnce(new Error('Test error'));

        return store.dispatch(serviceActions.fetchService(123)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        });
    });

    it('should create START_SERVICE_CONFIGS_LOADING', () => {
        expect(serviceActions.startServiceConfigsLoading(123, 0, 20)).toEqual(serviceActionsFixtures.START_SERVICE_CONFIGS_LOADING);
    });

    it('should create END_SERVICE_CONFIGS_LOADING', () => {
        expect(serviceActions.endServiceConfigsLoading(123, SERVICE_CONFIGS)).toEqual(serviceActionsFixtures.END_SERVICE_CONFIGS_LOADING);
    });

    it('should create START_SERVICE_CONFIGS_LOADING and END_SERVICE_CONFIGS_LOADING', () => {
        const store = mockStore(storeFixtures.SERVICE_EMPTY);

        fetch.mockResponseOnce(JSON.stringify(SERVICE_CONFIGS));

        return store.dispatch(serviceActions.fetchServiceConfigs(123, 0, 20)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        });
    });

    it('should create START_SERVICE_CONFIGS_LOADING and SET_ERRORS', () => {
        const store = mockStore(storeFixtures.SERVICE_EMPTY);

        fetch.mockRejectOnce(new Error('Test error'));

        return store.dispatch(serviceActions.fetchServiceConfigs(123, 0, 20)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        });
    });

    it('should create START_SERVICE_CONFIG_LOADING', () => {
        expect(serviceActions.startServiceConfigLoading(123, 456)).toEqual(serviceActionsFixtures.START_SERVICE_CONFIG_LOADING);
    });

    it('should create END_SERVICE_CONFIG_LOADING', () => {
        expect(serviceActions.endServiceConfigLoading(123, 456, SERVICE_CONFIG)).toEqual(serviceActionsFixtures.END_SERVICE_CONFIG_LOADING);
    });

    it('should create START_SERVICE_CONFIG_LOADING and END_SERVICE_CONFIG_LOADING', () => {
        const store = mockStore(storeFixtures.SERVICE_EMPTY);

        fetch.mockResponseOnce(JSON.stringify(SERVICE_CONFIG));

        return store.dispatch(serviceActions.fetchConfig(123, 456)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        });
    });

    it('should create START_SERVICE_LOADING and SET_ERRORS', () => {
        const store = mockStore(storeFixtures.SERVICE_EMPTY);

        fetch.mockRejectOnce(new Error('Test error'));

        return store.dispatch(serviceActions.fetchConfig(123, 456)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        });
    });

    it('should create START_SERVICE_LOADING and END_SERVICE_CONFIG_LOADING with empty data', () => {
        const store = mockStore(storeFixtures.SERVICE_EMPTY);

        fetch.mockResponseOnce('Not found', {status: 404});

        return store.dispatch(serviceActions.fetchConfig(123, 456)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        });
    });

    it('should create START_SERVICE_CONFIG_APPLYING with target=active', () => {
        expect(serviceActions.startServiceConfigApplying(123, 234, {
            target: 'active',
            oldConfigId: 235
        })).toEqual(serviceActionsFixtures.START_SERVICE_CONFIG_APPLYING);
    });

    it('should create END_SERVICE_CONFIG_APPLYING with target=active', () => {
        expect(serviceActions.endServiceConfigApplying(123, 234, {
            target: 'active',
            oldConfigId: 235
        })).toEqual(serviceActionsFixtures.END_SERVICE_CONFIG_APPLYING);
    });

    it('should create START_SERVICE_CONFIG_APPLYING and END_SERVICE_CONFIG_APPLYING with target=active', () => {
        const store = mockStore(storeFixtures.SERVICE_WITH_CONFIGS);

        fetch.mockResponseOnce(JSON.stringify({}));

        return store.dispatch(serviceActions.applyConfig(123, 234, {
            target: 'active',
            oldConfigId: 235
        })).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        });
    });

    it('should create START_SERVICE_CONFIG_APPLYING and SET_ERRORS with target=active', () => {
        const store = mockStore(storeFixtures.SERVICE_WITH_CONFIGS);

        fetch.mockRejectOnce(new Error('Test error'));

        return store.dispatch(serviceActions.applyConfig(123, 234, {
            target: 'active',
            oldConfigId: 235
        })).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        });
    });

    it('should create START_SERVICE_SETUP_LOADING', () => {
        expect(serviceActions.startServiceSetupLoading(123)).toEqual(serviceActionsFixtures.START_SERVICE_SETUP_LOADING);
    });

    it('should create END_SERVICE_SETUP_LOADING', () => {
        expect(serviceActions.endServiceSetupLoading(123, SETUP)).toEqual(serviceActionsFixtures.END_SERVICE_SETUP_LOADING);
    });

    it('should create START_SERVICE_SETUP_LOADING and END_SERVICE_SETUP_LOADING', () => {
        const store = mockStore(storeFixtures.SERVICE);

        // fetch.mockResponseOnce(JSON.stringify(SETUP));

        return store.dispatch(serviceActions.fetchServiceSetup(123)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        });
    });

    it('should create START_SERVICE_AUDIT_LOADING', () => {
        expect(serviceActions.startServiceAuditLoading('yandex_afisha', 0, 20)).toEqual(serviceActionsFixtures.START_SERVICE_AUDIT_LOADING);
    });

    it('should create START_SERVICE_AUDIT_LOADING with default params', () => {
        expect(serviceActions.resetServiceAudit('yandex_afisha')).toEqual(serviceActionsFixtures.RESET_SERVICE_AUDIT);
    });

    it('should create END_SERVICE_AUDIT_LOADING', () => {
        expect(serviceActions.endServiceAuditLoading('yandex_afisha', SERVICE_AUDIT)).toEqual(serviceActionsFixtures.END_SERVICE_AUDIT_LOADING);
    });

    it('should create START_SERVICE_AUDIT_LOADING and END_SERVICE_AUDIT_LOADING', () => {
        const store = mockStore(storeFixtures.SERVICE_EMPTY);

        fetch.mockResponseOnce(JSON.stringify(SERVICE_AUDIT));

        return store.dispatch(serviceActions.fetchServiceAudit('yandex_afisha', 0, 20)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        });
    });

    it('should create START_SERVICE_AUDIT_LOADING and SET_ERRORS', () => {
        const store = mockStore(storeFixtures.SERVICE_EMPTY);

        fetch.mockRejectOnce(new Error('Test error'));

        return store.dispatch(serviceActions.fetchServiceAudit('yandex_afisha', 0, 20)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        });
    });

    it('should create START_SERVICE_CONFIG_ARCHIVED_SETTING', () => {
        expect(serviceActions.startServiceConfigArchivedSetting('yandex_afisha', 345, true)).toEqual(serviceActionsFixtures.START_SERVICE_CONFIG_ARCHIVED_SETTING);
    });

    it('should create END_SERVICE_CONFIG_ARCHIVED_SETTING', () => {
        expect(serviceActions.endServiceConfigArchivedSetting('yandex_afisha', 345, true)).toEqual(serviceActionsFixtures.END_SERVICE_CONFIG_ARCHIVED_SETTING);
    });

    it('should create START_SERVICE_CONFIG_ARCHIVED_SETTING and END_SERVICE_CONFIG_ARCHIVED_SETTING', () => {
        const store = mockStore(storeFixtures.SERVICE_EMPTY);

        fetch.mockResponseOnce(JSON.stringify(SERVICE_CONFIG_ARCHIVED));

        return store.dispatch(serviceActions.setArchivedConfig('yandex_afisha', 345, true)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        });
    });

    it('should create START_SERVICE_CONFIG_ARCHIVED_SETTING and SET_ERRORS', () => {
        const store = mockStore(storeFixtures.SERVICE_EMPTY);

        fetch.mockRejectOnce(new Error('Test error'));

        return store.dispatch(serviceActions.setArchivedConfig('yandex_afisha', 345, true)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        });
    });

    it('should create START_SERVICE_COMMENT', () => {
        expect(serviceActions.startServiceComment(123)).toEqual(serviceActionsFixtures.START_SERVICE_COMMENT);
    });

    it('should create END_SERVICE_COMMENT', () => {
        expect(serviceActions.endServiceComment(123, 'comment')).toEqual(serviceActionsFixtures.END_SERVICE_COMMENT);
    });
});
