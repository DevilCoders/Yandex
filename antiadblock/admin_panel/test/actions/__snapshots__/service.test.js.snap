// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`service actions should create START_SERVICE_AUDIT_LOADING and END_SERVICE_AUDIT_LOADING 1`] = `
Array [
  Object {
    "limit": 20,
    "offset": 0,
    "serviceId": "yandex_afisha",
    "type": "START_SERVICE_AUDIT_LOADING",
  },
  Object {
    "audit": Object {
      "items": Array [
        Object {
          "action": "config_active",
          "date": "Mon, 12 Feb 2018 16:19:23 GMT",
          "id": 1,
          "params": Object {
            "config_comment": "wertwer",
            "config_id": 3,
            "old_config_id": 1,
          },
          "service_id": "yandex_afisha",
          "user_id": 204409962,
        },
        Object {
          "action": "config_test",
          "date": "Mon, 12 Feb 2018 16:46:39 GMT",
          "id": 2,
          "params": Object {
            "config_comment": "Initial config",
            "config_id": 1,
            "old_config_id": null,
          },
          "service_id": "yandex_afisha",
          "user_id": 204409962,
        },
        Object {
          "action": "config_active",
          "date": "Mon, 12 Feb 2018 16:51:53 GMT",
          "id": 3,
          "params": Object {
            "config_comment": "Initial config",
            "config_id": 1,
            "old_config_id": 3,
          },
          "service_id": "yandex_afisha",
          "user_id": 204409962,
        },
        Object {
          "action": "config_test",
          "date": "Mon, 12 Feb 2018 16:56:02 GMT",
          "id": 4,
          "params": Object {
            "config_comment": "wertwer",
            "config_id": 3,
            "old_config_id": null,
          },
          "service_id": "yandex_afisha",
          "user_id": 204409962,
        },
        Object {
          "action": "config_active",
          "date": "Mon, 12 Feb 2018 16:57:02 GMT",
          "id": 5,
          "params": Object {
            "config_comment": "wertwer",
            "config_id": 3,
            "old_config_id": 1,
          },
          "service_id": "yandex_afisha",
          "user_id": 204409962,
        },
        Object {
          "action": "config_test",
          "date": "Mon, 12 Feb 2018 17:18:51 GMT",
          "id": 6,
          "params": Object {
            "config_comment": "Initial config",
            "config_id": 1,
            "old_config_id": null,
          },
          "service_id": "yandex_afisha",
          "user_id": 204409962,
        },
      ],
      "total": 6,
    },
    "serviceId": "yandex_afisha",
    "type": "END_SERVICE_AUDIT_LOADING",
  },
]
`;

exports[`service actions should create START_SERVICE_AUDIT_LOADING and SET_ERRORS 1`] = `
Array [
  Object {
    "limit": 20,
    "offset": 0,
    "serviceId": "yandex_afisha",
    "type": "START_SERVICE_AUDIT_LOADING",
  },
  Object {
    "errors": Array [
      "Произошла ошибка. Попробуйте ещё раз.",
    ],
    "id": "global",
    "type": "SET_ERRORS",
  },
]
`;

exports[`service actions should create START_SERVICE_CONFIG_APPLYING and END_SERVICE_CONFIG_APPLYING with target=active 1`] = `
Array [
  Object {
    "configId": 234,
    "labelId": 123,
    "options": Object {
      "oldConfigId": 235,
      "target": "active",
    },
    "type": "START_SERVICE_CONFIG_APPLYING",
  },
  Object {
    "configId": 234,
    "labelId": 123,
    "options": Object {
      "oldConfigId": 235,
      "target": "active",
    },
    "type": "END_SERVICE_CONFIG_APPLYING",
  },
]
`;

exports[`service actions should create START_SERVICE_CONFIG_APPLYING and SET_ERRORS with target=active 1`] = `
Array [
  Object {
    "configId": 234,
    "labelId": 123,
    "options": Object {
      "oldConfigId": 235,
      "target": "active",
    },
    "type": "START_SERVICE_CONFIG_APPLYING",
  },
  Object {
    "errors": Array [
      "Произошла ошибка. Попробуйте ещё раз.",
    ],
    "id": "global",
    "type": "SET_ERRORS",
  },
]
`;

exports[`service actions should create START_SERVICE_CONFIG_ARCHIVED_SETTING and END_SERVICE_CONFIG_ARCHIVED_SETTING 1`] = `
Array [
  Object {
    "archived": true,
    "configId": 345,
    "serviceId": "yandex_afisha",
    "type": "START_SERVICE_CONFIG_ARCHIVED_SETTING",
  },
  Object {
    "archived": true,
    "configId": 345,
    "serviceId": "yandex_afisha",
    "type": "END_SERVICE_CONFIG_ARCHIVED_SETTING",
  },
]
`;

exports[`service actions should create START_SERVICE_CONFIG_ARCHIVED_SETTING and SET_ERRORS 1`] = `
Array [
  Object {
    "archived": true,
    "configId": 345,
    "serviceId": "yandex_afisha",
    "type": "START_SERVICE_CONFIG_ARCHIVED_SETTING",
  },
  Object {
    "errors": Array [
      "Произошла ошибка. Попробуйте ещё раз.",
    ],
    "id": "global",
    "type": "SET_ERRORS",
  },
]
`;

exports[`service actions should create START_SERVICE_CONFIG_LOADING and END_SERVICE_CONFIG_LOADING 1`] = `
Array [
  Object {
    "configId": 456,
    "serviceId": 123,
    "type": "START_SERVICE_CONFIG_LOADING",
  },
  Object {
    "config": Object {
      "comment": "stand alone a config",
      "date": "2018-01-10T11:12:13",
      "id": 456,
      "status": "inactive",
    },
    "configId": 456,
    "serviceId": 123,
    "type": "END_SERVICE_CONFIG_LOADING",
  },
]
`;

exports[`service actions should create START_SERVICE_CONFIGS_LOADING and END_SERVICE_CONFIGS_LOADING 1`] = `
Array [
  Object {
    "labelId": 123,
    "limit": 20,
    "offset": 0,
    "type": "START_SERVICE_CONFIGS_LOADING",
  },
  Object {
    "configs": Object {
      "items": Array [
        Object {
          "comment": "made a config",
          "date": "2018-01-10T11:12:13",
          "id": 234,
          "statuses": Array [
            Object {
              "comment": "",
              "status": "preview",
            },
          ],
        },
        Object {
          "comment": "made a config",
          "date": "2018-01-10T11:12:13",
          "id": 235,
          "statuses": Array [
            Object {
              "comment": "",
              "status": "active",
            },
          ],
        },
        Object {
          "comment": "made a config",
          "date": "2018-01-10T11:12:13",
          "id": 236,
          "statuses": Array [
            Object {
              "comment": "",
              "status": "preview",
            },
          ],
        },
        Object {
          "comment": "made a config",
          "date": "2018-01-10T11:12:13",
          "id": 237,
          "statuses": Array [
            Object {
              "comment": "",
              "status": "preview",
            },
          ],
        },
        Object {
          "comment": "made a config",
          "date": "2018-01-10T11:12:13",
          "id": 238,
          "statuses": Array [
            Object {
              "comment": "",
              "status": "active",
            },
          ],
        },
        Object {
          "comment": "made a config",
          "date": "2018-01-10T11:12:13",
          "id": 239,
          "statuses": Array [],
        },
        Object {
          "comment": "made a config",
          "date": "2018-01-10T11:12:13",
          "id": 240,
          "statuses": Array [
            Object {
              "comment": "",
              "status": "active",
            },
          ],
        },
        Object {
          "comment": "made a config",
          "date": "2018-01-10T11:12:13",
          "id": 241,
          "statuses": Array [
            Object {
              "comment": "",
              "status": "active",
            },
          ],
        },
        Object {
          "comment": "made a config",
          "date": "2018-01-10T11:12:13",
          "id": 242,
          "statuses": Array [
            Object {
              "comment": "",
              "status": "active",
            },
          ],
        },
        Object {
          "comment": "made a config",
          "date": "2018-01-10T11:12:13",
          "id": 243,
          "statuses": Array [
            Object {
              "comment": "",
              "status": "active",
            },
          ],
        },
        Object {
          "comment": "made a config",
          "date": "2018-01-10T11:12:13",
          "id": 244,
          "statuses": Array [
            Object {
              "comment": "",
              "status": "preview",
            },
          ],
        },
        Object {
          "comment": "made a config",
          "date": "2018-01-10T11:12:13",
          "id": 245,
          "statuses": Array [],
        },
        Object {
          "comment": "made a config",
          "date": "2018-01-10T11:12:13",
          "id": 246,
          "statuses": Array [],
        },
        Object {
          "comment": "made a config",
          "date": "2018-01-10T11:12:13",
          "id": 247,
          "statuses": Array [
            Object {
              "comment": "",
              "status": "active",
            },
          ],
        },
        Object {
          "comment": "made a config",
          "date": "2018-01-10T11:12:13",
          "id": 248,
          "statuses": Array [
            Object {
              "comment": "",
              "status": "active",
            },
          ],
        },
        Object {
          "comment": "made a config",
          "date": "2018-01-10T11:12:13",
          "id": 249,
          "statuses": Array [
            Object {
              "comment": "",
              "status": "preview",
            },
          ],
        },
        Object {
          "comment": "made a config",
          "date": "2018-01-10T11:12:13",
          "id": 250,
          "statuses": Array [
            Object {
              "comment": "",
              "status": "active",
            },
          ],
        },
        Object {
          "comment": "made a config",
          "date": "2018-01-10T11:12:13",
          "id": 251,
          "statuses": Array [
            Object {
              "comment": "",
              "status": "active",
            },
          ],
        },
        Object {
          "comment": "made a config",
          "date": "2018-01-10T11:12:13",
          "id": 252,
          "statuses": Array [
            Object {
              "comment": "",
              "status": "active",
            },
          ],
        },
        Object {
          "comment": "made a config",
          "date": "2018-01-10T11:12:13",
          "id": 253,
          "statuses": Array [
            Object {
              "comment": "",
              "status": "active",
            },
          ],
        },
      ],
      "total": 60,
    },
    "labelId": 123,
    "type": "END_SERVICE_CONFIGS_LOADING",
  },
]
`;

exports[`service actions should create START_SERVICE_CONFIGS_LOADING and SET_ERRORS 1`] = `
Array [
  Object {
    "labelId": 123,
    "limit": 20,
    "offset": 0,
    "type": "START_SERVICE_CONFIGS_LOADING",
  },
  Object {
    "errors": Array [
      "Произошла ошибка. Попробуйте ещё раз.",
    ],
    "id": "global",
    "type": "SET_ERRORS",
  },
]
`;

exports[`service actions should create START_SERVICE_LOADING and END_SERVICE_CONFIG_LOADING 1`] = `
Array [
  Object {
    "serviceId": 123,
    "type": "START_SERVICE_LOADING",
  },
  Object {
    "errors": Array [
      "Произошла ошибка. Попробуйте ещё раз.",
    ],
    "id": "global",
    "type": "SET_ERRORS",
  },
]
`;

exports[`service actions should create START_SERVICE_LOADING and END_SERVICE_CONFIG_LOADING with empty data 1`] = `
Array [
  Object {
    "configId": 456,
    "serviceId": 123,
    "type": "START_SERVICE_CONFIG_LOADING",
  },
  Object {
    "errors": Array [
      "Произошла ошибка. Попробуйте ещё раз.",
    ],
    "id": "global",
    "type": "SET_ERRORS",
  },
]
`;

exports[`service actions should create START_SERVICE_LOADING and SET_ERRORS 1`] = `
Array [
  Object {
    "serviceId": 123,
    "type": "START_SERVICE_LOADING",
  },
  Object {
    "errors": Array [
      "Произошла ошибка. Попробуйте ещё раз.",
    ],
    "id": "global",
    "type": "SET_ERRORS",
  },
]
`;

exports[`service actions should create START_SERVICE_LOADING and SET_ERRORS 2`] = `
Array [
  Object {
    "configId": 456,
    "serviceId": 123,
    "type": "START_SERVICE_CONFIG_LOADING",
  },
  Object {
    "errors": Array [
      "Произошла ошибка. Попробуйте ещё раз.",
    ],
    "id": "global",
    "type": "SET_ERRORS",
  },
]
`;

exports[`service actions should create START_SERVICE_SETUP_LOADING and END_SERVICE_SETUP_LOADING 1`] = `
Array [
  Object {
    "serviceId": 123,
    "type": "START_SERVICE_SETUP_LOADING",
  },
  Object {
    "serviceId": 123,
    "setup": Object {
      "js_inline": "(func () { /* here goes awesome JS code */})(\\"<PARTNER-ID>\\")",
      "nginx_config": "here 
 goes 
 config",
      "token": "AABBCCDDEEFF....",
    },
    "type": "END_SERVICE_SETUP_LOADING",
  },
]
`;
